src/Sets/Ellipsoid.jl

```diff
@@ -56,39 +73,34 @@ julia> σ(ones(3), E)
  0.5773502691896258
  0.5773502691896258
 ```
-
-A two-dimensional ellipsoid with given center and shape matrix:
-
-```julia
-julia> E = Ellipsoid(ones(2), Diagonal([2.0, 0.5]))
-Ellipsoid{Float64}([1.0, 1.0], [2.0 0.0; 0.0 0.5])
-```
 """
 struct Ellipsoid{N<:AbstractFloat} <: AbstractCentrallySymmetric{N}
     center::AbstractVector{N}
     shape_matrix::AbstractMatrix{N}
 
     # default constructor with dimension check
-    function Ellipsoid{N}(c::AbstractVector{N},
-                          Q::AbstractMatrix{N}) where {N<:AbstractFloat}
-        @assert length(c) == checksquare(Q)
-        @assert isposdef(Q) "an ellipsoid's shape matrix must be positive " *
-            "definite"
+    function Ellipsoid(c::AbstractVector{N},
+                       Q::AbstractMatrix{N};
+                       check_posdef::Bool=true) where {N<:AbstractFloat}
+
+        @assert length(c) == checksquare(Q) "the length of the center and the size "
+            "of the shape matrix do not match; they are $(length(c)) and $(size(Q)) respectively"
+
+        if check_posdef
+            isposdef(Q) || throw(ArgumentError("an ellipsoid's shape matrix " *
+                                    "must be positive definite"))
+        end
         return new{N}(c, Q)
     end
 end
 
 isoperationtype(::Type{<:Ellipsoid}) = false
 isconvextype(::Type{<:Ellipsoid}) = true
 
-# convenience constructor without type parameter
-Ellipsoid(c::AbstractVector{N}, Q::AbstractMatrix{N}) where {N<:AbstractFloat} =
-    Ellipsoid{N}(c, Q)
-
 # convenience constructor for ellipsoid centered in the origin
-Ellipsoid(Q::AbstractMatrix{N}) where {N<:AbstractFloat} =
-    Ellipsoid(zeros(N, size(Q, 1)), Q)
```

Comment:
This change is unrelated and has been done in #1946 already. It is not clear to me whether we want to have that. Can you comment in #1946?

Reply:
#1946 has other changes too; i didn't have time to consider them yet..

Reply:
But you're right; i shouldn't have removed that constructor. It's just weird to have `Ellipsoid{N}` in line 82  and then again `new{N}(...)`  below.

Reply:
> #1946 has other changes too

No, there are just some consequences. That is why I think we should discuss these changes before removing stuff. You can undo the changes here or we wait until we reached an agreement.